src/lib/core/compatibility/compatibility.ts

```diff
@@ -171,7 +171,7 @@ export class CompatibilityModule {
   }
 
   constructor(@Optional() @Inject(DOCUMENT) document: any) {
-    if (isDevMode() && typeof document && !document.doctype) {
+    if (isDevMode() && document && !document.doctype) {
```

Comment:
Won't just `document` by itself potentially be an issue for Universal? `typeof document == 'object'`?